File: 102.binary-tree-level-order-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
	results := [][]int{}
	nodes := []*TreeNode{root}
	newNodes := []*TreeNode{}
	for len(nodes) != 0 {
		currentVal := []int{}
		for i := range nodes {
			if nodes[i] != nil {
				newNodes = append(newNodes, nodes[i].Left, nodes[i].Right)
				currentVal = append(currentVal, nodes[i].Val)
			}
		}
		nodes = newNodes
		newNodes = []*TreeNode{}
		if len(currentVal) > 0 {
			results = append(results, append([]int{}, currentVal...))
		}
	}
	return results
}

File: 102.binary-tree-level-order-traversal.py
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        q1 = [root]
        q2 = []
        result = []
        while len(q1) > 0:
            cur = []
            for t in q1:
                if t is None:
                    continue
                q2.append(t.left)
                q2.append(t.right)
                cur.append(t.val)
            if len(cur) > 0:
                result.append(cur)
            q1 = q2
            q2 = []
        return result
